src/patternfly/components/Popover/popover-title-text.hbs

```diff
@@ -1,6 +1,22 @@
-<span class="pf-c-popover__title-text{{#if popover-title-text--modifier}} {{popover-title-text--modifier}}{{/if}}"
+<{{#if popover-title--type}}{{popover-title--type}}{{else}}h1{{/if}} class="pf-c-popover__title-text{{#if popover-title-text--modifier}} {{popover-title-text--modifier}}{{/if}}"
   {{#if popover-title-text--attribute}}
     {{{popover-title-text--attribute}}}
   {{/if}}>
+    {{#if popover--IsAlert}}
+    <span class="pf-u-screen-reader">
```

Comment:
If we allow the consumer to pass their own title in, would it then be up to them to include the sr text in their title to keep it in the heading element used for the title? We're including the sr text in the heading for them now:

* https://github.com/patternfly/patternfly-react/blob/01c2451bcb6af1dc7767d86e6884c509b6b41891/packages/react-core/src/components/Popover/PopoverHeader.tsx#L39-L45
* https://github.com/patternfly/patternfly-react/blob/01c2451bcb6af1dc7767d86e6884c509b6b41891/packages/react-core/src/components/Modal/ModalBoxTitle.tsx#L56-L69

I wonder if we could include the sr text in the title wrapper instead, then we'll include it regardless the title used.

Reply:
@mcoker I don't know if that still works ok. Could this be a follow on issue so that this can move forward?

Reply:
@srambach yeah, though since it can effect the way the title updates are made in react (and other components with this structure), I wonder if we can decide now. If not, I think we can merge it and follow up. @thatblindgeye or @nicolethoen do you have an opinion on this?

We're updating the structure (in other components, too) so that the "title" element is just a container (`<div>`) for the title to be used. In that element, you can use the "title-text" element that PF provides to get the preferred element (heading, div, etc) and title text styling that comes from design, or you can use a `<Title>` (or whatever) component instead if you want to manage your own title.

Currently if the popover has an alert status, we include sr-only text in the popover title heading element. Since a user can potentially opt-out of our new title heading element (title-text) and use their own title, we can only predictably add sr-only text to the heading element if they use our title-text element. Seems like we could go a couple of ways with that.

1. Document that alert status popovers (or other components with this structure) need sr-only text in the title heading element, and we include that in the title-text element. If a user is providing their own title, they'll need to be sure and include the sr-only text themselves. This is probably my preference.
2. Move the sr-only text out of the heading element, and as the first thing in the title container (`<div>`). I'm not sure of the implications of that since the text is no longer in the heading element.

Reply:
I think I would agree with your preference, @mcoker, that we document it and if a consumer wants to use their own custom thing, it's up to them to make sure they take any additional considerations. At least with VO, having it as part of the `title-text` has it announced as a single text content ("Default alert: Default popover title"), whereas moved outside of they become more separated.

One thing unrelated to this conversation is the `pf-u-screen-reader` class being applied. I believe we're trying to update usage of that class to just `pf-screen-reader`